### PR TITLE
Release instance id lock on interrupt

### PR DESCRIPTION
**Pull request by jemoreira**

### CODE REVIEW

File: base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp [hunk 1]
```diff
@@ -755,6 +755,14 @@ Result<cvd::Response> CvdStartCommandHandler::Handle(
                         "`cvd reset` to ensure a clean state";
         }
 
+        for (const auto& instance: group.Instances()) {
+          InstanceLockFileManager ilfm;
+          if (instance.id() > 0) {
+            // Do this before the id is reset in the next block
+            ilfm.RemoveLockFile(instance.id());
+          }
+        }
```
Comment (Databean) [inline base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp:764]: If this is deleting the locks indicating some instance ids are in use, should those instances be shut down too?

**Review by Databean — CHANGES_REQUESTED** [review]

File: base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp (hunk 1 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp:764]: that happens on line 750: `subprocess_waiter_.Interrupt()`

Comment (Databean) [reply base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp:764]: That `Interrupt()` will `SIGKILL` the process, does that guarantee all the instance subprocesses are shut down in `--daemon` mode?

Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp:764]: In most cases it does. There is a small window in which the subprocess gets killed and the instances remain, that's why there is warning below.

**Review by Databean — APPROVED** [review]

**Pull request merged**